assets/src/blocks/Covers/useCovers.js

```diff
@@ -39,11 +77,52 @@ export const useCovers = ({ post_types, tags, cover_type, initialRowsLimit, post
     }
   };
 
+  const slideCovers = (direction, rowNumber) => {
+    if (!coversContainer.current) {
+      return;
+    }
+
+    const covers = coversContainer.current.querySelector('.covers');
+
+    if (!covers) {
+      return;
+    }
+
+    const isRTL = document.querySelector('html').dir === 'rtl';
+    // We need to account for the spacing between covers,
+    // in all styles that spacing is 24px for screens >= 992px.
+    let scrollOffset = (isRTL ? -1 : 1) * (covers.offsetWidth + 24);
+
+    // For the Take Action covers because of the box-shadow we had to add
+    // extra padding to the sides (5px each) so we need to account for them.
+    if (cover_type === COVERS_TYPES.takeAction) {
+      scrollOffset += (isRTL ? 1 : -1) * 10;
+    }
+
+    if (direction) {
+      const initialScrollPosition = covers.scrollLeft;
+      covers.scrollLeft = initialScrollPosition + (direction === 'next' ? scrollOffset : -scrollOffset);
+      setRow(direction === 'next' ? row + 1 : row - 1);
+    } else if (rowNumber && rowNumber !== row) {
+      covers.scrollLeft = (rowNumber - 1) * scrollOffset;
+      setRow(rowNumber);
+    }
+  };
+
   useEffect(() => {
     if (!noLoading) {
       loadCovers();
     }
-  }, [ cover_type, post_types, tags, posts ]);
+  }, [cover_type, post_types, tags, posts, layout]);
+
+  useEffect(() => {
+    updateRowCoversAmount();
+
+    if (layout !== COVERS_LAYOUTS.carousel) {
+      window.addEventListener('resize', updateRowCoversAmount);
+      return () => window.removeEventListener('resize', updateRowCoversAmount);
```

Comment:
Just an opinion, I would prefer to not `return` this removed listener. maybe this will be better 
```
return () => { window.removeEventListener('resize', updateRowCoversAmount); }
```

Reply:
Hm, I think it's fine as is, we use similar notations in several other places 🙂 Such as [CarouselHeaderFrontend](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/blocks/CarouselHeader/CarouselHeaderFrontend.js)